Dockerfile

```diff
@@ -0,0 +1,85 @@
+FROM ubuntu:20.04
+
+ENV TERM="xterm-color"
+ARG USER_ID=999
+ARG GROUP_ID=999
+ARG USER=docker
+ARG PWD=docker
+ARG HOME="/home/$USER"
+
+# Create our group and user so we can login as ourself. Useful for ssh.
+RUN groupadd -g $GROUP_ID $USER && \
+    useradd -g $GROUP_ID -m -s /bin/bash -u $USER_ID $USER
+
+# Suppresses prompt for timezone
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Yocto dependencies
+RUN apt-get update
+RUN apt-get -y upgrade
+RUN apt-get -y install gawk make wget tar bzip2 gzip python3 unzip perl patch \
+    diffutils diffstat git cpp gcc texinfo chrpath \
+    ccache socat \
+    python3-pexpect findutils file cpio python python3-pip \
+    python3-jinja2 xterm bash
+
+# Unable to locate packages. These were likely packaged with other packages
+# gcc-c++  glibc-devel perl-Data-Dumper perl-Text-ParseWords perl-Thread-Queue perl-bignum
+# which xz SDL-devel python3-GitPython
+
+# Other stuff that might be helpful
+RUN apt-get -y install vim curl ca-certificates
+
+# Install and setup google repo
+RUN mkdir -p $HOME/.bin
+RUN echo '\n\
+export PATH="$HOME/.bin:$PATH"\n\
+\n' >> ${HOME}/.bashrc
+RUN curl https://storage.googleapis.com/git-repo-downloads/repo > ~/.bin/repo
+RUN chmod a+rx ~/.bin/repo
+
```

Comment:
This should be done after you switch to $USER.  I think all user configurations should group together for that $USER.  Also, see commend about adduser above.